tool/state-processor/src/staking/mod.rs

```diff
@@ -117,6 +149,124 @@ impl<S> Processor<S> {
+				// The inflation for current era doesn't happen yet, skip.
+				continue;
+			}
+
+			let era_payout = {
+				let mut p = *eras_validator_reward.get(&era_raw).unwrap();
+
+				p.adjust();
+
+				p as f64
+			};
+
+			for (stash, exposure) in stakers {
+				let controller = array_bytes::bytes2hex("0x", bonded.get(&stash).unwrap());
+				let ledger = ledgers.get(&controller).unwrap();
+
+				if ledger.claimed_rewards.contains(&era) {
+					// Already claimed.
+					continue;
+				}
+
+				let era_reward_points = era_reward_points.get(&era_raw).unwrap();
+				let Some(validator_reward_points) = era_reward_points.individual.get(&ledger.stash).cloned() else {
+					// No reward points.
+					continue;
+				};
+				let total_reward_points = era_reward_points.total;
+				let validator_total_reward_part =
+					validator_reward_points as f64 / total_reward_points as f64;
+				let validator_total_payout = era_payout * validator_total_reward_part;
+				let validator_prefs =
+					eras_validator_prefs.get(&era_raw).unwrap().get(&stash).unwrap();
+				// `Perbill` to `f64`.
+				let validator_commission = validator_prefs.commission as f64 / 1_000_000_000_f64;
+				let validator_commission_payout = validator_total_payout * validator_commission;
+				let validator_leftover_payout =
+					validator_total_payout - validator_commission_payout;
+				let validator_exposure_part =
+					exposure.own_power as f64 / exposure.total_power as f64;
+				let validator_staking_payout = validator_leftover_payout * validator_exposure_part;
+				let validator_payout = validator_commission_payout + validator_staking_payout;
+
+				payout += validator_payout as Balance;
+
+				exposure.others.into_iter().for_each(|nominator| {
+					let nominator_exposure_part =
+						nominator.power as f64 / exposure.total_power as f64;
+					let nominator_payout = validator_leftover_payout * nominator_exposure_part;
+
+					payout += nominator_payout as Balance;
+
+					let nominator = array_bytes::bytes2hex("0x", nominator.who);
+					let nominator_payout = nominator_payout as Balance;
+
+					log::trace!("pay `{nominator_payout}` to `nominator({nominator})`");
+
+					self.shell_state.transfer(staking_pallet_account, &nominator, nominator_payout);
+				});
+
+				let validator_payout = validator_payout as Balance;
+
+				log::trace!("pay `{validator_payout}` to `validator({stash})`");
+
+				self.shell_state.transfer(staking_pallet_account, &stash, validator_payout);
+			}
+		}
+
+		{
+			let mut a = AccountInfo::default();
+
+			self.shell_state.get_value(
+				b"System",
+				b"Account",
+				&blake2_128_concat_to_string(array_bytes::hex2array_unchecked::<_, 20>(
+					staking_pallet_account,
+				)),
+				&mut a,
+			);
+
+			log::info!("`staking_pallet_account_balance({})`", a.data.free);
+			log::info!("                        `payout({payout})`");
+
+			assert!(
+				a.data.free > payout,
+				"`staking_pallet_account_balance` must be greater then `payout`"
+			);
+		}
```

Comment:
How do we deal with the remaining part in the `PalletId(da/staki)` account? This one is different from the one in darwinia 2.0 staking pallet.

```rs
PalletId(*b"dar/stak").into_account_truncating()
```

Reply:
This is unimplemented yet, right?

Reply:
#270